pyhms/sprout/sprout_filters.py

```diff
@@ -93,6 +96,40 @@ def __call__(
         return candidates
 
 
+class MahalanobisFarEnough(DemeLevelCandidatesFilter):
+    def __init__(self, percentile: float) -> None:
+        super().__init__()
+        self.percentile = percentile
+        self._threshold: float | None = None
+
+    def __call__(
+        self,
+        candidates: dict[AbstractDeme, DemeCandidates],
+        tree,
+    ) -> dict[AbstractDeme, DemeCandidates]:
+        for deme in candidates.keys():
+            if self._threshold is None:
+                self._set_threshold(deme)
+            child_siblings = [sibling for sibling in tree.levels[deme.level + 1]]
+            child_seeds = candidates[deme].individuals
+            for sibling in child_siblings:
+                if not isinstance(sibling, CMADeme):
+                    continue
```

Comment:
So currently this sprout filter is intended for levels with CMA demes only?

Reply:
Yes, but we can easily extend this to other demes by estimating their covariance matrices.